### PR TITLE
Readme 2.0

### PR DESCRIPTION
**Pull request by phatblat**
An attempt to address #30. The "Subproject" section still needs to be reworked, especially for iOS, but I wanted to get feedback on the direction I'm going, especially on the "Features" section. I started it as a quick list of functionality, but wanted to incorporate links so passers-by can quickly get to the relevant API  or examples. Now I'm debating how much value that provides over having broken links as things change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,59 +4,143 @@
...
+* Network: 
+[Clone](ObjectiveGit/GTRepository.h#L231), 
+[Fetch](ObjectiveGit/GTRepository+RemoteOperations.h#L34),
+[Push](ObjectiveGit/GTRepository+RemoteOperations.h#L67),
+Pull (in progress #464)
+* Transports: HTTPs, SSH, local filesystem
+
+Not all libgit2 features are available, but if you run across something missing, please consider [contributing a pull request](#contributing)!
+
+Many classes in the ObjectiveGit API wrap a C struct from libgit2 and expose the underlying data and operations using Cocoa ideoms. The underlying libgit2 types are prefixed with `git_` and are often accessible via a property so that your application can take advantage of the libgit2 API directly.
```
Comment (mdiep) [inline README.md]: Typo here: `idioms`

**Comment by mdiep** [comment]
I like it! :+1:

File: README.md (hunk 1 above)
Comment (phatblat) [inline README.md]: Thx

**Comment by joshaber** [comment]
>I wanted to get feedback on the direction I'm going, especially on the "Features" section. I started it as a quick list of functionality, but wanted to incorporate links so passers-by can quickly get to the relevant API or examples. Now I'm debating how much value that provides over having broken links as things change.

I worry about that too, and whether we'll remember to keep it up to date :disappointed:

**Comment by phatblat** [comment]
This is ready for review now.

**Comment by joshaber** [comment]
Fantastic :metal:

**Pull request merged**